docs/DeveloperGuide.md

```diff
@@ -1,34 +1,814 @@
+
+Here is the sequence diagram to better illustrate the lifecycle of a browse command.
+
+![Browse Sequence Diagram](images/Browse-SequenceDiagram.png) <br/>
+*Figure 13: Browse Sorted Sequence Diagram*
+
+<br/>
+
+#### 4.2.2 Design Consideration
+Here are some various design considerations that was taken when implementing the `browse` feature.
+
+Aspect: **How should the program handle the sorted list**
+
+The first design consideration was how the sorting should be carried out. The main issues here are the time and storage complexity.
+
+| Approach | Pros | Cons  |
+| --- | --- | --- |
+| 1. Leaving the list unsorted                    | - No complexity and fastest approach  | - List will be unsorted and may cause confusion to users |
+| 2. Resorting the list again                     | - The list will be back into its original form before browsing    | - May hinder performance as resorting could take time <br/> - Requires altering of the main list |
+| 3. Cloning a duplicate animeData object to sort | - The list will be back to its original form <br/> - The main list will not be affected at all | - Expensive operation that will require large storage and time complexity |
+
+Currently, the 2nd approach is being used with the following rationale. While the 1st approach is the fastest, the consequence of 
+leaving the main list unsorted is too great and may produce a lot of uncertain results as well as confuse the user. 
+Although the 3rd approach provides the best benefit, its complexity may end up violating the project’s memory limit constraint if the list is large. 
+Therefore, the 2nd approach was chosen, as its performance cost outweighs the other approaches cons.  
+
+Aspect: **Should the program use an interactive or static browsing approach**
+
+The second design consideration was how to carry out the page by page browsing as shown above. 
+The main issue was the cohesiveness between components.
+
+| Approach | Pros | Cons  |
+| --- | --- | --- |
+| 1. Interactive browsing, users can `flip` pages and `pick` to view specific anime | - Fluid and seamless browsing session. <br/>- Very good usability. | - Would require `BrowseCommand` to get user input. |
+| 2. Static browsing, users specify the page they want to access. | - Completely decoupled from Ui component <br/>- Allows for browse to be more precise in finding what the user wants |  - Not as seamless as the first approach but still usable. |
+
+Though the 1st approach could have created a more authentic browsing feature it would not fit the requirements as well as the 
+2nd approach. The 2nd approach allows for more precise browsing of pages means that more experienced users are able to utilise the 
+tool quicker and to the same effect as the first approach. As a result, the 2nd approach was chosen as a better fit of the requirements 
+and in favour of having an application that is highly object-oriented.
```

Comment:
Good to see that all aspects for the browsing have been well thought out and considered before making your decision on the implementation to go for. Well documented as well!